File: 830.较大分组的位置.py
#
# @lc app=leetcode.cn id=830 lang=python3
#
# [830] 较大分组的位置
#

# @lc code=start
class Solution:
    def largeGroupPositions(self, S: str) -> List[List[int]]:
        ans = []
        i = 0
        for j in range(len(S)):
            if j == len(S) - 1 or S[j] != S[j+1]:
                if j - i + 1>= 3:
                    ans.append([i, j])
                i = j + 1
        return ans
                
        
# @lc code=end

File: 830.较大分组的位置.cpp
/*
 * @lc app=leetcode.cn id=830 lang=cpp
 *
 * [830] 较大分组的位置
 */

// @lc code=start
class Solution {
public:
    vector<vector<int>> largeGroupPositions(string s) {
        vector<vector<int>>res;
        int slow = 0, fast = 0;
        while(fast < s.size()){
            while(s[slow] == s[fast]){
                fast++;
            }
            if(slow+1 == fast)
                slow = fast;
            else{
                if(fast - slow >= 3){
                    res.push_back({slow, fast-1});
                }
                slow = fast;
            }
        }
        return res;
    }
};
// @lc code=end

